uitools/toolkitcpp/cpp/Esri/ArcGISRuntime/Toolkit/PieChartPopupMediaItem.h

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ *  Copyright 2012-2025 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+#ifndef ESRI_ARCGISRUNTIME_TOOLKIT_PIECHARTPOPUPMEDIAITEM_H
+#define ESRI_ARCGISRUNTIME_TOOLKIT_PIECHARTPOPUPMEDIAITEM_H
+
+// Qt headers
+#include <QJsonArray>
+#include <QObject>
+
+// Other headers
+#include "PopupMediaItem.h"
+
+class QPieSlice;
+
+namespace Esri::ArcGISRuntime {
+
+namespace Toolkit {
+
+class PieChartPopupMediaItem : public PopupMediaItem
+{
+  Q_OBJECT
+  Q_PROPERTY(QList<QPieSlice*> pieSlices READ pieSlices NOTIFY pieChartPopupMediaItemChanged)
+
+public:
+  explicit PieChartPopupMediaItem(PopupMedia* popupMedia, const QJsonArray& jsonColorArray, QObject* parent = nullptr);
+  ~PieChartPopupMediaItem() override = default;
+
+  QList<QPieSlice*> pieSlices();
```

Comment:
I mentioned this on one property, but I think it could apply everywhere. Is there any case for other C++ code to call this? If not, make this method private so that its sole access is via the property in QML.